### PR TITLE
[NAE-1521] Authentication Providers

### PR DESCRIPTION
**Pull request by machacjozef**
# Description

<Please include a summary of the changes and which issue is fixed. Please also include relevant links and special instructions if applicable.>

Implements [NAE-1521]

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

Manually

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |   Linux Mint 20.3        |
| Runtime             |  Java 11        |
| Dependency Manager  |  Maven 3.8.4         |
| Framework version   |  Spring Boot 2.6.7      |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @mladoniczky 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1521]: https://netgrif.atlassian.net/browse/NAE-1521?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/netgrif/application/engine/auth/service/AfterRegistrationAuthService.java [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
-    }
-
-    @Override
-    public void logoutAfterRegistrationFinished() {
-        SecurityContextHolder.getContext().setAuthentication(null);
-    }
-
-
-}
+//import com.netgrif.application.engine.auth.service.interfaces.IAfterRegistrationAuthService;
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/auth/service/AfterRegistrationAuthService.java:4]: why is this class not removed?

File: src/main/java/com/netgrif/application/engine/auth/service/interfaces/IAfterRegistrationAuthService.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package com.netgrif.application.engine.auth.service.interfaces;
 
-public interface IAfterRegistrationAuthService {
-    void authenticateWithUsernameAndPassword(String username, String password);
-    void logoutAfterRegistrationFinished();
-}
+//public interface IAfterRegistrationAuthService {
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/auth/service/interfaces/IAfterRegistrationAuthService.java:3]: why is this interface not removed?

File: src/main/java/com/netgrif/application/engine/configuration/NAESecurityConfiguration.java [hunk 3]
```diff
@@ -42,13 +37,13 @@
...
+
 @Slf4j
 @Configuration
 @Controller
 @EnableWebSecurity
-@Order(SecurityProperties.BASIC_AUTH_ORDER)
-@ConditionalOnExpression("!${nae.server.security.static.enabled} && !${nae.ldap.enabled}")
-public class SecurityConfiguration extends AbstractSecurityConfiguration {
+@Order(SecurityProperties.DEFAULT_FILTER_ORDER)
+public class NAESecurityConfiguration extends AbstractSecurityConfiguration {
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/configuration/NAESecurityConfiguration.java:46]: why `NAESecurity...` and not `NaeSecurity`? The property classes used in this class use the latter naming?

File: src/main/java/com/netgrif/application/engine/configuration/security/PublicAuthenticationFilter.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public PublicAuthenticationFilter(ProviderManager authenticationManager, Anonymo
                                       Authority anonymousAuthority, String[] urls, String[] exceptions, IJwtService jwtService,
                                       IUserService userService) {
         this.authenticationManager = authenticationManager;
-        this.authenticationManager.getProviders().add(provider);
+//        this.authenticationManager.getProviders().add(provider);
```
Comment (minop) [inline src/main/java/com/netgrif/application/engine/configuration/security/PublicAuthenticationFilter.java:44]: why was this not removed?

**Review by minop — DISMISSED** [review]

**Review by minop — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**